### PR TITLE
feat(UI): Add BA tree widget

### PR DESCRIPTION
**Pull request by Thebarda**
## Description

This adds the widget that allow the federated component on BAM side to be displayed:
- Add the widget
- Move federated modules and widgets atom to ui-context
- Add support for preload script from modules
- Add additional resources atom on ui-context

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [x] 24.04.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

## Summary by CodeRabbit

- **New Features**
	- Added multiple language translations for terms related to business activity diagrams and resource statuses.
	- Introduced new atoms for managing additional resources and federated modules/widgets.
	- Enhanced tree graph components to support different link types and introduced user-select styling.
	- Implemented new functionalities in Zoom components, allowing dynamic ID handling for better customization.
	- Added a new widget "Business Activity Diagram" for visually representing business activity hierarchies.

- **Documentation**
	- Updated documentation to include new export functionalities and component properties.

- **Tests**
	- Expanded test coverage for tree display, Zoom functionalities, and dashboard configurations.

- **Refactor**
	- Updated atom usage in widget selection and input management to reflect new naming conventions.

### CODE REVIEW

**Comment by Thebarda** [comment]
@coderabbitai read centreon/www/front_src/src/Dashboards/SingleInstancePage/Dashboard/AddEditWidget/WidgetProperties/Inputs/Resources/useResources.ts and generate cypress component testing code.

**Comment by Thebarda** [comment]
https://github.com/coderabbitai read centreon/www/front_src/src/Dashboards/SingleInstancePage/Dashboard/AddEditWidget/WidgetProperties/Inputs/Resources/Resources.tsx and generate cypress component testing code.

File: centreon/www/widgets/src/centreon-widget-groupmonitoring/properties.json [hunk 1]
```diff
@@ -16,35 +16,41 @@
...
+          "id": "2",
+          "name": "Warning"
+        },
+        {
+          "id": "4",
+          "name": "Pending"
+        },
+        {
+          "id": "5",
+          "name": "Success (OK & Up)"
```
Comment (Thebarda) [inline centreon/www/widgets/src/centreon-widget-groupmonitoring/properties.json:43]:
```suggestion
          "name": "Success (OK/Up)"

```

**Review by dmyios — DISMISSED** [review]

**Review by Noha-ElAbrouki — APPROVED** [review]

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 2]
```diff
@@ -18916,3 +18912,27 @@ msgstr "Standard"
...
+msgstr "Succès (OK)"
+
+msgid "Link types"
+msgstr "Types de lien"
+
+msgid "Curve"
+msgstr "Courbe"
+
+msgid "Line"
+msgstr "Ligne"
```
Comment (Thebarda) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:18938]:
```suggestion
msgstr "Ligne"

msgid "Step"
msgstr "Étape"
```

**Review by smau-centreon — DISMISSED** [review]

**Review by dmyios — DISMISSED** [review]

**Review by MedWassCh — DISMISSED** [review]

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 3]
```diff
@@ -18917,6 +18913,33 @@ msgstr "Standard"
...
+
+msgid "Step"
+msgstr "Étape"
+
```
Comment (Thebarda) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:18942]:
```suggestion
```

**Review by dmyios — APPROVED** [review]

**Review by smau-centreon — APPROVED** [review]

**Review by MedWassCh — APPROVED** [review]

**Review by Noha-ElAbrouki — APPROVED** [review]

**Pull request merged**